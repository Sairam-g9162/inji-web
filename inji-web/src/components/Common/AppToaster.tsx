import React from "react";
import { ToastContainer, toast } from "react-toastify";
import { useSelector } from "react-redux";
import { RootState } from "../../types/redux";
import { isRTL } from "../../utils/i18n";

const CustomToast = ({ message }: { message: string }) => (
    <div data-testid="custom-toast">{message}</div>
);

export const AppToaster: React.FC = () => {
    const language = useSelector((state: RootState) => state.common.language);
    const positionClass = isRTL(language) ? "top-left" : "top-right";

    // Example of triggering a toast
    React.useEffect(() => {
        toast(<CustomToast message="Test message" />);
    }, []);

    return (
        <div className={`toast-container ${positionClass}`} data-testid="toast-container">
            <ToastContainer
                position={isRTL(language) ? "top-left" : "top-right"}
                autoClose={1000}
                hideProgressBar
                newestOnTop
                closeOnClick
                rtl={isRTL(language)}
                icon={<React.Fragment />}
                pauseOnFocusLoss
                draggable
                pauseOnHover
                style={{ width: '400px' }}
                theme="colored"
            />
        </div>
    );
};
