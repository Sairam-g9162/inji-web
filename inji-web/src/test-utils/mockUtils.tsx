import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { reduxStore } from '../redux/reduxStore';
import { useDispatch } from 'react-redux';

export const mockUseTranslation = () => {
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
            t: (key: string) => key,
        }),
    }));
};

export const mockUseNavigate = () => {
    const mockNavigate = jest.fn();
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: mockNavigate,
    }));
};

export const mockUseSearchcredentials = () =>{
    jest.mock('../components/Credentials/SearchCredential', () => ({
        SearchCredential: () => <></>,
    }));
};

export const mockUseSelector = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useSelector: jest.fn(),
    }));
};

export const mockUsegetObjectForCurrentLanguage = () =>{
    jest.mock('../utils/i18n', () => ({
        getObjectForCurrentLanguage: jest.fn(),
    }));
};

export const wrapUnderRouter = (children: React.ReactNode) => {
    return <Router>{children}</Router>;
};

export const mockUseDispatch = () =>{
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useDispatch:jest.fn(),
    }));
};

export const mockUseSpinningLoader = () =>{
    jest.mock('../components/Common/SpinningLoader', () => ({
        SpinningLoader: () => <></>,
    }));
};
export const mockUseLanguageSelector = () =>{
    jest.mock('../components/Common/LanguageSelector', () => ({
        LanguageSelector: () => <></>,
    }));
};

interface RenderWithProviderOptions extends Omit<RenderOptions, 'queries'> {}

const renderWithProvider = (ui: ReactElement, options?: RenderWithProviderOptions) => {
    return render(
        <Provider store={reduxStore}>
            <Router>
                {ui}
            </Router>
        </Provider>,
        options
    );
};

export * from '@testing-library/react';
export { renderWithProvider };
