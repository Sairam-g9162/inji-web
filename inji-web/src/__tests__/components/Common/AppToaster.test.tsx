import React from 'react';
import {  screen } from '@testing-library/react';
import { AppToaster } from '../../../components/Common/AppToaster';
import { reduxStore } from '../../../redux/reduxStore'; // Assuming the store is named reduxStore
import { renderWithProvider } from '../../../test-utils/mockUtils';

describe('Testing the Layout of AppToaster', () => {
    test('Check if the layout is matching with the snapshots for English language', () => {
        const { asFragment } = renderWithProvider(<AppToaster />);
        reduxStore.dispatch({ type: 'SET_LANGUAGE', payload: 'en' });
        expect(asFragment()).toMatchSnapshot();
    });

    test('Check if the layout is matching with the snapshots for Arabic language', () => {
        const { asFragment } = renderWithProvider(<AppToaster />);
        reduxStore.dispatch({ type: 'SET_LANGUAGE', payload: 'ar' });
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Testing the Functionality of AppToaster', () => {
    test('Check if functionalities works properly with English language', () => {
        renderWithProvider(<AppToaster />);
        reduxStore.dispatch({ type: 'SET_LANGUAGE', payload: 'en' });
        const customToast = screen.getByTestId('AppToaster-custom-toast');
        expect(customToast).toHaveTextContent('AppToaster test message');
    });

    test('Check if functionalities works properly with Arabic language', () => {
        renderWithProvider(<AppToaster />);
        reduxStore.dispatch({ type: 'SET_LANGUAGE', payload: 'ar' });
        const customToast = screen.getByTestId('AppToaster-custom-toast');
        expect(customToast).toHaveTextContent('AppToaster test message');
    });
});
