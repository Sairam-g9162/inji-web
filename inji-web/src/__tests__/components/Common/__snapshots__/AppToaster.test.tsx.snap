// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layout of AppToaster Check if the layout is matching with the snapshots for Arabic language 1`] = `
<DocumentFragment>
  <div
    class="text-center pt-20 pb-10 top-right"
    data-testid="Apptoaster-outer-container"
  >
    <div
      class="Toastify"
    >
      <div
        class="Toastify__toast-container Toastify__toast-container--top-right"
        style="width: 400px;"
      >
        <div
          class="Toastify__toast Toastify__toast-theme--colored Toastify__toast--default Toastify__toast--close-on-click Toastify--animate Toastify__bounce-enter--top-right"
          data-in="true"
          id="2"
        >
          <div
            class="Toastify__toast-body"
            role="alert"
          >
            <div
              class="Toastify__toast-icon Toastify--animate-icon Toastify__zoom-enter"
            />
            <div>
              <div
                data-testid="AppToaster-custom-toast"
              >
                AppToaster test message
              </div>
            </div>
          </div>
          <button
            aria-label="close"
            class="Toastify__close-button Toastify__close-button--colored"
            type="button"
          >
            <svg
              aria-hidden="true"
              viewBox="0 0 14 16"
            >
              <path
                d="M7.71 8.23l3.75 3.75-1.48 1.48-3.75-3.75-3.75 3.75L1 11.98l3.75-3.75L1 4.48 2.48 3l3.75 3.75L9.98 3l1.48 1.48-3.75 3.75z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <div
            class="Toastify__progress-bar--wrp"
            data-hidden="true"
          >
            <div
              class="Toastify__progress-bar--bg Toastify__progress-bar-theme--colored Toastify__progress-bar--default"
            />
            <div
              aria-hidden="true"
              aria-label="notification timer"
              class="Toastify__progress-bar Toastify__progress-bar--animated Toastify__progress-bar-theme--colored Toastify__progress-bar--default"
              role="progressbar"
              style="animation-duration: 1000ms; animation-play-state: paused;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the Layout of AppToaster Check if the layout is matching with the snapshots for English language 1`] = `
<DocumentFragment>
  <div
    class="text-center pt-20 pb-10 top-right"
    data-testid="Apptoaster-outer-container"
  >
    <div
      class="Toastify"
    >
      <div
        class="Toastify__toast-container Toastify__toast-container--top-right"
        style="width: 400px;"
      >
        <div
          class="Toastify__toast Toastify__toast-theme--colored Toastify__toast--default Toastify__toast--close-on-click Toastify--animate Toastify__bounce-enter--top-right"
          data-in="true"
          id="1"
        >
          <div
            class="Toastify__toast-body"
            role="alert"
          >
            <div
              class="Toastify__toast-icon Toastify--animate-icon Toastify__zoom-enter"
            />
            <div>
              <div
                data-testid="AppToaster-custom-toast"
              >
                AppToaster test message
              </div>
            </div>
          </div>
          <button
            aria-label="close"
            class="Toastify__close-button Toastify__close-button--colored"
            type="button"
          >
            <svg
              aria-hidden="true"
              viewBox="0 0 14 16"
            >
              <path
                d="M7.71 8.23l3.75 3.75-1.48 1.48-3.75-3.75-3.75 3.75L1 11.98l3.75-3.75L1 4.48 2.48 3l3.75 3.75L9.98 3l1.48 1.48-3.75 3.75z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <div
            class="Toastify__progress-bar--wrp"
            data-hidden="true"
          >
            <div
              class="Toastify__progress-bar--bg Toastify__progress-bar-theme--colored Toastify__progress-bar--default"
            />
            <div
              aria-hidden="true"
              aria-label="notification timer"
              class="Toastify__progress-bar Toastify__progress-bar--animated Toastify__progress-bar-theme--colored Toastify__progress-bar--default"
              role="progressbar"
              style="animation-duration: 1000ms; animation-play-state: paused;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
